src/amo/components/Category/index.js

```diff
@@ -67,11 +68,17 @@ export function mapStateToProps(state, ownProps) {
     { ...state.search.filters, page: parsePage(state.search.page) },
     { ...filters, page: queryParams.page },
   );
+
+  const hasFetched = state.categories.hasFetched;
+  const loading = state.categories.loading || state.search.loading;
```

Comment:
This looks like the right thing to do. Once you make this `loading` prop effective I believe that you won't need `hasFetched`.